app/jobs/migrate_notifications_job.rb

```diff
@@ -0,0 +1,63 @@
+class MigrateNotificationsJob < ApplicationJob
+  queue_as :default
+
+  def perform(*args)
+    # Do something later
+  end
+end
+class MigrateNotificationsJob < ApplicationJob
+  queue_as :default
+
+  BATCH_SIZE = 10
+  LIMIT = 100
+
+  # Define the Notification model to access the old table
+  class Notification < ApplicationRecord
+    self.inheritance_column = nil
+  end
+  
+
+  def perform(*args)  
+    total_processed = 0
+
+    last_processed_at = get_last_processed_at
+
+    # Process notifications in batches  
+    Notification.order(created_at: :desc).where("created_at > ?", last_processed_at).limit(LIMIT).find_in_batches(batch_size: BATCH_SIZE) do |batch|
+      batch.each do |notification|
+        return if total_processed >= LIMIT
+        migrate_notification(notification)
+        total_processed += 1      
+      end    
+    end
+  end
+
+  private
+
+  def get_last_processed_at
+    last_event = Noticed::Event.order(created_at: :asc).first
+    last_event ? last_event.created_at : Time.at(0)
```

Comment:
```suggestion
    last_event = last_event&.created_at || Notification.first.created_at - 1.hour
```
I'd prefer to do `Notification.first.created_at - 1.hour` over `Time.at(0)` just to make it a bit clearer that we are getting the date BEFORE the first notification was created.